lib/ec2imgutils/ec2deprecateimg.py

```diff
@@ -173,17 +174,38 @@ def _get_images_to_deprecate(self):
 
     # ---------------------------------------------------------------------
     def _set_deletion_date(self):
-        """Set the date when the deprecation perios expires"""
-        now = datetime.datetime.now()
-        expire = now + dateutil.relativedelta.relativedelta(
+        """Set the date when the deprecation period expires"""
+        dep_date = None
+        if self.deprecation_date:
+            try:
+                dep_date = datetime.datetime.strptime(
+                    self.deprecation_date, '%Y%m%d')
+            except Exception as e:
+                msg = 'The deprecation date, "%s", is not valid.' \
+                    % self.deprecation_date
+                raise EC2DeprecateImgException(msg) from e
+        else:
+            dep_date = datetime.datetime.now()
+            self._set_deprecation_date('')
+        expire = dep_date + dateutil.relativedelta.relativedelta(
             months=+self.deprecation_period)
         self.deletion_date = self._format_date(expire)
 
     # ---------------------------------------------------------------------
-    def _set_deprecation_date(self):
-        """Set the deprecation day in the YYYYMMDD format"""
-        now = datetime.datetime.now()
-        self.deprecation_date = self._format_date(now)
+    def _set_deprecation_date(self, deprecation_date):
```

Comment:
Should default to '', i.e. `deprecation_date=''` then....

Reply:
Done!